normandy/recipes/filters.py

```diff
@@ -194,6 +194,122 @@ def capabilities(self):
         return set()
 
 
+class PrefCompareFilter(BaseFilter):
+    """Match based on a user's pref having a particular value.
+
+    .. attribute:: type
+
+        ``pref``
+
+    .. attribute:: value
+
+        string, boolean, or number.
+
+        :example: ``true`` or ``"default"`` or "10"
+
+    .. attribute:: comparison
+
+        Options are ``equal``, ``not_equal``, ``greater_than``,
+        ``less_than``, ``greater_than_equal`` and ``less_than_equal``.
+    """
+
+    type = "pref"
+    pref = serializers.CharField()
+    value = serializers.JSONField()
+    comparison = serializers.CharField()
+
+    def to_jexl(self):
+        comparison = self.initial_data["comparison"]
+        value = self.initial_data["value"]
+        pref = self.initial_data["pref"]
+
+        if comparison == "equal":
+            symbol = "=="
+        elif comparison == "not_equal":
+            symbol = "!="
+        elif comparison == "greater_than":
+            symbol = "<"
```

Comment:
Isn't `<` the symbol for less than, not greater? Why'd you flip them?

Reply:
I suspect this was because of my suggestion that value should be on the left of the operator, thus looking at how `value` compares to `preferenceValue` instead of the other way around. This change was largely to allow the `in` to make sense. But maybe we should flip that back and `in` should be a special case that is renamed to `contains`?

Reply:
i'm thinking about the UI now and realizing you probably want to enter the pref name first, then the comparison type and then the value you're comparing to. so maybe the original order of comparing `preferenceValue` to `value` makes more sense.

i.e. for `>`, `>=`, `<`, `<=`, `==`:
```py
return f"'{pref}'|preferenceValue {symbol} {json.dumps(value)}"
```

but for `contains`:
```py
return f"{json.dumps(value)} in '{pref}'|preferenceValue"
```

Reply:
Oh 🤦‍♂️. I totally missed the swapped arguments below. I think the way it is now, where comparison words and the comparison operators are flipped is going to cause a lot of confusion (see: me an hour ago). This works, I suppose, but it seems very strange to me. At the very least, it should have a comment explaining what's going on.

Reply:
change made.

Reply:
> Oh 🤦‍♂️. I totally missed the swapped arguments below. I think the way it is now, where comparison words and the comparison operators are flipped is going to cause a lot of confusion (see: me an hour ago). This works, I suppose, but it seems very strange to me. At the very least, it should have a comment explaining what's going on.

yeah i also did a bad job explaining that i didn't think the operators should be flipped. but i think where we've landed is the best solution. thanks @uhlissuh!